### DISCUSSION TITLE
Redirect rtmp stream to other origin

### DISCUSSION THREAD

**Discussion by Pavel-Mishalov** [OvenMediaLabs/OvenMediaEngine#1513] [Help] [answered]
**Describe**
Hi.
When I receive an AdmissionWebhooks request for permission to open an rtmp stream in the response, I give the other origin host on which the broadcast recording was started using the new_url parameter.
But the stream remains on the same origin server and recording is in the stopped state

**To Reproduce**
1. [Server.xml.j2](https://github.com/AirenSoft/OvenMediaEngine/files/14076429/Server.txt)
2. Use OBS 27.2.4
3. Stream did not open on the origin server that I response 'new_url' in AddmissionWebhook

**Comment by nums** [comment]
RTMP redirection is certainly not supported with OBS. Many encoders do not support RTMP redirection

**Reply by Pavel-Mishalov** [reply]
Of course, I understand that obs does not redirect the request. Can OME to redirect or proxy the stream to another origin server on which it started streaming and we started recording?

**Reply by nums** [reply, marked as answer]
Yes, you can do this with Ome API => https://airensoft.gitbook.io/ovenmediaengine/rest-api/v1/virtualhost/application/push

**Reply by Pavel-Mishalov** [reply]
Yes, it's probably for me. Thank you so much for your help.
I am also interested, when we receive a request in AddmissionWebhook with the incoming direction to create new stream and in the new_url response, how does it behave?)

example request:
```
POST /configured/target/url/ HTTP/1.1
Content-Length: 325
Content-Type: application/json
Accept: application/json
X-OME-Signature: f871jd991jj1929jsjd91pqa0amm1
{
  "client": {
      "address": "128.0.0.176",
      "port": 64465
  },
  "request": {
      "direction": "incoming",
     "protocol": "RTMP",
     "status": "opening",
     "time": "2024-02-04T11:53:31.642+00:00",
     "url": "rtmp://stream-001.dev-streams.tv:1935/live/uwIAxg3vIO4"
  }
}
```

example response:
```
HTTP/1.1 200 OK
Content-Length: 102
Content-Type: application/json
Connection: Closed
{
  "allowed": true,
  "new_url": "rtmp://stream-002.dev-streams.tv:1935/medium/uwIAxg3vIO4",
  "lifetime": 0,
  "reason": "Allow stream from OBS"
}
```

In the example, the stream was created in this "medium" application, I also wanted to change the host, since the documentation describes that we can change all url parameters except the schema, port and file. However, the stream was created on the host `stream-002.dev-streams.tv `

![image](https://github.com/AirenSoft/OvenMediaEngine/assets/23746493/a51b72e7-7721-4d00-b874-f6d24ea483c0)

**Reply by getroot** [reply]
This is a function to change to another stream within the same OME server. This is not a redirect function to send to another server.
In the manual "The host can only be changed to **another virtual host on the same server.**".
